spec/unit/middleware/service_broker_rate_limiter_spec.rb

```diff
@@ -60,27 +61,42 @@ module Middleware
           context 'when the path is /v2/*' do
             let(:path_info) { '/v2/service_instances' }
             it 'formats the response error in v2 format' do
-              _, _, body = middleware.call(user_env)
+              _, response_headers, body = middleware.call(user_env)
               json_body = JSON.parse(body.first)
               expect(json_body).to include(
                 'code' => 10016,
                 'description' => 'Service broker concurrent request limit exceeded',
                 'error_code' => 'CF-ServiceBrokerRateLimitExceeded',
               )
+              expect(response_headers['Retry-After']).to be_between(Time.now + broker_timeout * 0.5, Time.now + broker_timeout * 1.5)
             end
           end
 
           context 'when the path is /v3/*' do
             let(:path_info) { '/v3/service_instances' }
 
             it 'formats the response error in v3 format' do
-              _, _, body = middleware.call(user_env)
+              _, response_headers, body = middleware.call(user_env)
               json_body = JSON.parse(body.first)
               expect(json_body['errors'].first).to include(
                 'code' => 10016,
                 'detail' => 'Service broker concurrent request limit exceeded',
                 'title' => 'CF-ServiceBrokerRateLimitExceeded',
               )
+              expect(response_headers['Retry-After']).to be_between(Time.now + broker_timeout * 0.5, Time.now + broker_timeout * 1.5)
+            end
+          end
+
+          context 'when broker_client_timeout_seconds is reduced' do
+            let(:broker_timeout) { 3 }
+            let(:middleware) {
+              ServiceBrokerRequestCounter.instance.limit = concurrent_limit
+              ServiceBrokerRateLimiter.new(app, logger: logger, broker_timeout_seconds: broker_timeout)
+            }
+
+            it 'reduces the suggested delay in the Retry-After header' do
+              _, response_headers, _ = middleware.call(user_env)
+              expect(response_headers['Retry-After']).to be_between(Time.now + broker_timeout * 0.5, Time.now + broker_timeout * 1.5)
```

Comment:
Should this be rounded (floored / ceiled)? Or should we use a different `broker_timeout`, e.g. `4` for which it is more clear what to expect?